src/macro.jl

```diff
@@ -5,6 +5,13 @@ macro yield(expr=nothing)
   error("@yield macro outside a @resumable function!")
 end
 
+"""
+Macro if used in a `@resumable function` that delegates to `expr` otherwise throws an error.
+"""
+macro yield_from(expr=nothing)
+  error("@yield_from macro outside a @resumable function!")
+end
```

Comment:
Could we add a `@test_throws` for this, to make sure future changes do not bork it up?

Reply:
Added a test for that (using `eval()`, which wraps the actual error in a `LoadError`; not sure if there's a better way)